nuxeo-runtime/nuxeo-stream/src/test/java/org/nuxeo/lib/stream/tests/log/TestLog.java

```diff
@@ -862,8 +863,12 @@ public void testMixingCodec() throws Exception {
         appender.append(0, msg1);
 
         try (LogTailer<KeyValueMessage> tailer = manager.createTailer(GROUP, LogPartition.of(logName, 0), codecRead)) {
-            assertEquals(msg1, tailer.read(DEF_TIMEOUT).message());
-            assertEquals(null, tailer.read(SMALL_TIMEOUT));
+            try {
+                tailer.read(DEF_TIMEOUT);
+                fail("Try to read with a different codec should fail");
+            } catch (AvroRuntimeException e) {
+                // expected Decoding datum failed
```

Comment:
You could add `assertTrue(e.getMessage(), e.getMessage().contains("something relevant"))` to be complete.